src/connector/src/aws_utils.rs

```diff
@@ -238,9 +223,8 @@ impl AwsConfigV2 {
             .region(region.clone())
             .credentials_provider(cred_provider);
 
-        if let Some(endpoint_uri) = &self.endpoint {
-            let endpoint_val = Endpoint::immutable(endpoint_uri.uri.clone());
-            config_loader = config_loader.endpoint_resolver(endpoint_val);
+        if let Some(endpoint) = &self.endpoint {
+            config_loader = config_loader.endpoint_url(endpoint);
```

Comment:
Have we verified that this change doesn't break the souece?

Reply:
lgtm so far, lets trigger main cron ci after merge